compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -3974,15 +3978,15 @@ fn hint_missing_borrow<'tcx>(
 
     if !to_borrow.is_empty() {
         err.multipart_suggestion_verbose(
-            "consider borrowing the argument",
+            "consider adjusting the signature so it borrows its argument",
             to_borrow,
             Applicability::MaybeIncorrect,
```

Comment:
```
    if !to_borrow.is_empty() {
        let msg = "consider adjusting the signature so it borrows its arguments";
        let msg = if to_borrow.len() > 1 { msg } else { &msg[..msg.len() - 1] };
        err.multipart_suggestion_verbose(
            msg
            to_borrow,
            // ...
```

Pluralization would be pretty easy, not sure if we do it many places

Reply:
I may as well just migrate this to a struct diagnostic.